apisupport/apisupport.ant/src/org/netbeans/modules/apisupport/project/Evaluator.java

```diff
@@ -889,7 +890,11 @@ private String mergePaths(Set<String> cnbs, boolean test,String testtype,File te
         return cps.toString();
     }
 
-    private static final Map<String, String> limitModulesCache = new HashMap<>();
+    /**
+     * cache shared between Evaluator instances.
+     */
+    private static final Map<String, String> limitModulesCache = new ConcurrentHashMap<>();
+
     private static String getLimitModules(String javacRelease) {
         return limitModulesCache.computeIfAbsent(javacRelease, release -> {
```

Comment:
on second thought: this probably could be a plain old synchronized map? The keys are javac release versions, so it won't use the per-item lock feature of the CMH very often.